extract.py

```diff
@@ -1,179 +1,143 @@
+from pathlib import Path
 import argparse
+import textract
 import PyPDF2
-from pathlib import Path
+import signal
+import nxttxt
 import time
 
 
-# "Modular" functions (used multiple times each)
-def get_mode(input_mode: str):
-    output_mode: int = 0o777
-    if not input_mode:
-        return output_mode
-    try:
-        int(input_mode)
-    except ValueError:
-        print(input_mode, "is not a valid unix permission level.")
-        exit(1)
-    else:
-        output_mode = int("0o" + input_mode)
-    return output_mode
-
-
-def enumerate_duplicate_paths(start_path: str):
-    test_path = str(Path(start_path).with_suffix(''))
-    start_suffixes = Path(start_path).suffixes
-    end_suffix = "".join(start_suffixes)
-
-    ends_in_a_number: bool
-
-    try:
-        last_number = test_path.split()[-1]
-        iteration_number = int(last_number)
-
-    except (ValueError, IndexError):
-        iteration_number = 1
-        ends_in_a_number = False
+# "Procedural" functions(called once each)
+def authenticate_instructional_validity(input_dir: str, output_dir: str) -> list:
+    print("Authenticate validity of entered paths...")
+    input_path = Path(input_dir)
+    if nxttxt.is_valid_path(input_path, True) and output_dir and (not nxttxt.is_valid_path(output_dir,
+                                                                                           True)):
+        result = [False, "The specified output path is not valid.", nxttxt.exceptions.InvalidPath]
+    elif not nxttxt.is_valid_path(input_path, False):
+        result = [False, "The specified input path is not valid.", nxttxt.exceptions.InvalidPath]
+    elif not nxttxt.is_valid_path(input_path, True):
+        result = [False, "The specified input path is not a directory.", nxttxt.exceptions.PathIsNotADirectory]
```

Comment:
Why not raise the errors here?

Reply:
Just a personal organization thing. I prefer to have fewer raise statements, have them grouped, and have them near the problem because I find it easier to debug. Do you think I should raise them there?